Libraries/LibCrypto/OpenSSL.h

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2024, Altomani Gianluca <altomanigianluca@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Error.h>
+
+#include <openssl/err.h>
+#include <openssl/evp.h>
+
+namespace Crypto {
+
+#define OPENSSL_TRY_PTR(...)                                                                 \
+    ({                                                                                       \
+        auto* _temporary_result = (__VA_ARGS__);                                             \
+        if (!_temporary_result) [[unlikely]] {                                               \
+            auto err = ERR_get_error();                                                      \
+            VERIFY(err);                                                                     \
+            auto* err_message = ERR_error_string(err, nullptr);                              \
+            return Error::from_string_view(StringView { err_message, strlen(err_message) }); \
```

Comment:
I don't think this is safe to do, IIRC `ERR_error_string` will overwrite the static buffer on every call?

Reply:
You are right, I should make a new buffer beforehand.

Reply:
Fixed, not sure if there's a better way to make the buffer and pass it around.